### PR TITLE
Handle tracebacks when user sets --volume-id for fresh volumes

### PR DESCRIPTION
**Pull request by vatsa287**
Signed-off-by: Shree Vatsa N <vatsa@kadalu.io>

### CODE REVIEW

File: mgr/src/server/plugins/volume_utils.cr [hunk 1]
```diff
@@ -92,17 +92,33 @@ def validate_volume_create(req)
         FileUtils.rmdir storage_unit.path unless storage_unit_pre_exists
       end
 
+      # Avoid scenario of user creating the volume with user's --volume-id instead of
+      # random value set by 'moana' when path is created by agent.
+      if !storage_unit_pre_exists && req.volume_id
```
Comment (aravindavk) [inline mgr/src/server/plugins/volume_utils.cr]: Why not? This can be allowed.

File: mgr/src/server/plugins/volume_utils.cr [hunk 2]
```diff
@@ -92,17 +92,33 @@ def validate_volume_create(req)
...
+        return NodeResponse.new(false,{
+          "error": "Cannot use --volume-id option with non pre-existing path #{storage_unit.path}"}.to_json
+        )
+      end
+
       xattr_vol_id = get_xattr(storage_unit.path, VOLUME_ID_XATTR_NAME) if storage_unit_pre_exists
```
Comment (aravindavk) [inline mgr/src/server/plugins/volume_utils.cr:95]: Good to add rescue here and handle error

File: mgr/src/server/plugins/volume_utils.cr (hunk 1 above)
Comment (vatsa287) [reply mgr/src/server/plugins/volume_utils.cr]: It may lead to "invalid string length" if not given correctly.

Comment (aravindavk) [reply mgr/src/server/plugins/volume_utils.cr]: Add validation to the API

```crystal
def valid_uuid?(val)
  UUID.new(val)
  true
rescue ArgumentError
  false
end
```

```crystal
if req.volume_id != "" && !valid_uuid?(req.volume_id)
   halt..
end
```

**Comment by aravindavk** [comment]
Rebase this PR to see the actual failure. Fixed the node response issue here https://github.com/kadalu/moana/pull/158

**Review by aravindavk — APPROVED** [review]

File: mgr/src/server/plugins/volume_utils.cr (hunk 1 above)
Comment (vatsa287) [reply mgr/src/server/plugins/volume_utils.cr]: done

**Pull request merged**